bitcoin/src/crypto/sighash.rs

```diff
@@ -472,6 +472,10 @@ impl EcdsaSighashType {
     pub fn to_u32(self) -> u32 { self as u32 }
 }
 
+impl From<EcdsaSighashType> for u32 {
+    fn from(s: EcdsaSighashType) -> u32 { s.to_u32() }
+}
```

Comment:
As I mentioned in the issue this might be unneeded and potentially even risky.

Reply:
How can converting *to* a u32 be risky?

But I agree it's really not needed. We have a conversion to `u8` and converting that to a `u32` is a very niche thing to do and doesn't need first-class API support.

Reply:
There was `Into<u32>` in the API which meant you could pass in either ECDSA or Schnorr sighahs into **ECDSA** method. Changing the signature to accept the strong type fixes the problem. (But also nobody in the entire ecosystem should have such API.)